### PR TITLE
New Package: sndio, aucatctl, gst-plugins-sndio

### PR DESCRIPTION
**Pull request by lanodan**
Been using them on my machines for few weeks, pushed it on the guru repository but needs to be in `::gentoo` for support of other programs like ffmpeg and mpd, which I'll address later on (need to change my patches from stable ebuilds to live ones)

### CODE REVIEW

File: acct-user/sndiod/sndiod-0.ebuild [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit acct-user
+
+DESCRIPTION="daemon user for sndio"
+ACCT_USER_ID="-1"
```
Comment (thesamesam) [inline acct-user/sndiod/sndiod-0.ebuild:9]: Not allowed in ::gentoo. Please reserve a UID/GID in the [API](https://github.com/gentoo/api-gentoo-org) repo (pick a UID/GID unlikely to be taken for a bit and make a PR there, link it here).

File: acct-user/sndiod/sndiod-0.ebuild [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ACCT_USER_GROUPS=( audio )
+ACCT_USER_HOME=/var/lib/sndio
```
Comment (thesamesam) [inline acct-user/sndiod/sndiod-0.ebuild:11]: See [this](http://michael.orlitzky.com/articles/gentoo_glep81_user_package_guidelines.xhtml) just to check you need USER_HOME.

File: media-plugins/gst-plugins-sndio/gst-plugins-sndio-1.8.0.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+SRC_URI="https://github.com/t6/gstreamer1-plugins-sndio/archive/v${PV}.tar.gz -> ${MY_P}.tar.gz"
+S="${WORKDIR}/${MY_P}"
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="
+	media-libs/gstreamer:1.0
+	media-sound/sndio:=
+	sys-devel/gettext
```
Comment (thesamesam) [inline media-plugins/gst-plugins-sndio/gst-plugins-sndio-1.8.0.1.ebuild:19]: gettext is usually only needed at build time (so BDEPEND).

File: media-sound/sndio/sndio-1.6.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+RDEPEND="
+	${DEPEND}
+	acct-user/sndiod
+"
+
+src_configure() {
+	./configure \
+		--prefix=/usr \
+		--libdir=/usr/$(get_libdir) \
+		--privsep-user=${PN}d \
```
Comment (thesamesam) [inline media-sound/sndio/sndio-1.6.0.ebuild:27]: I guess this doesn't need the user to exist or anything during build time?

**Comment by lanodan** [comment]
PR for `sndiod` ID: https://github.com/gentoo/api-gentoo-org/pull/279

File: media-sound/sndio/sndio-1.6.0.ebuild (hunk 4 above)
Comment (lanodan) [reply media-sound/sndio/sndio-1.6.0.ebuild:27]: Yeah, it only needs to exists as a username string and for the service.

**Comment by lanodan** [comment]
Bumped sndio to 1.7.0 and changed the UID to 390. The PR got closed as I claimed a wrong one and opening a PR isn't necessary (see <https://wiki.gentoo.org/wiki/Proxied_Maintainer_FAQ#Adding_acct-.7Bgroup.2Cuser.7D.2Fpackages_as_a_proxied_maintainer>).

**Comment by thesamesam** [comment]
I'd actually love to try use this but it looks like sndio doesn't support Bluetooth. But that won't stop me testing, just using it daily. This is on my list to look at.

**Comment by lanodan** [comment]
I think it should work with bluetooth but I haven't got the ability to test it (haven't got a bluetooth audio device that still works), I think you would need to use bluez-alsa.

**Comment by thesamesam** [comment]
Finally done, thank you for your patience!

**Pull request closed without merging**